processor/resourcedetectionprocessor/internal/aws/eks/detector.go

```diff
@@ -74,7 +88,20 @@ func (d *detector) Detect(ctx context.Context) (resource pcommon.Resource, schem
 	d.rb.SetCloudProvider(conventions.AttributeCloudProviderAWS)
 	d.rb.SetCloudPlatform(conventions.AttributeCloudPlatformAWSEKS)
 
-	return d.rb.Emit(), conventions.SchemaURL, nil
+	if d.ra.K8sClusterName.Enabled {
```

Comment:
Do we need to perform this check ourselves? It looks like the generated code does that for you inside `SetK8sClusterName`. I could see the benefit of doing the check, that way we only perform the query if it's enabled. It could confuse users if the see an error for cluster name but they don't have it enabled.

Reply:
Yes, you're hitting on the reasons why we added this check explicitly. The logic to get the k8s cluster name can take 10+ seconds, and will often hit errors when running with the default timeout (or always hit errors if user doesn't have proper permission). Adding this means the user won't have the long delay and unrelated errors if the cluster name is disabled.